service/storage_service.cc

```diff
@@ -3101,7 +3107,32 @@ future<std::map<gms::inet_address, float>> storage_service::effective_ownership(
         //
         // The call for get_range_for_endpoint is done once per endpoint
         const auto& tm = *erm->get_token_metadata_ptr();
-        const auto token_ownership = dht::token::describe_ownership(tm.sorted_tokens());
+        const auto token_ownership = std::invoke([&]() {
+            const replica::keyspace& ks = ss._db.local().find_keyspace(keyspace_name);
+            if (!ks.uses_tablets()) {
+                return dht::token::describe_ownership(tm.sorted_tokens());
+            } else {
```

Comment:
I need to find a better place for this logic. Also, I will double-check if it is correct. Right now the tests pass.

Basically, I had to calculate `token_ownership` on basis of tokens from tablet map instead of sorted tokens.